engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationProcessor.java

```diff
@@ -129,12 +130,20 @@ public void processRecord(
         eventKey, ProcessInstanceModificationIntent.MODIFIED, value, command);
   }
 
-  private Optional<Rejection> validateCommand(
+  private Either<Rejection, DeployedProcess> validateCommand(
       final TypedRecord<ProcessInstanceModificationRecord> command, final DeployedProcess process) {
     final var value = command.getValue();
+    final var activateInstructions = value.getActivateInstructions();
 
+    return validateElementExists(process, activateInstructions)
+        .map(valid -> process);
+  }
+
+
+  private Either<Rejection, DeployedProcess> validateElementExists(final DeployedProcess process,
+      final List<ProcessInstanceModificationActivateInstructionValue> activateInstructions) {
```

Comment:
❓ Why is the Right side of the return value of this method `DeployedProcess`? It looks unused and seems like a strange return type for this validation method. Perhaps we just want to return `?`. We could do the same as in ProcessInstanceCreation, with the `VALID` constant.